src/Symfony/Bundle/FrameworkBundle/Logger/Logger.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Logger;
+
+use Psr\Log\AbstractLogger;
+use Psr\Log\InvalidArgumentException;
+use Psr\Log\LogLevel;
+
+/**
+ * Minimalist PSR-3 logger designed to write in stderr or any other stream.
+ *
+ * @internal
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+final class Logger extends AbstractLogger
+{
+    private static $levels = array(
+        LogLevel::DEBUG => 0,
+        LogLevel::INFO => 1,
+        LogLevel::NOTICE => 2,
+        LogLevel::WARNING => 3,
+        LogLevel::ERROR => 4,
+        LogLevel::CRITICAL => 5,
+        LogLevel::ALERT => 6,
+        LogLevel::EMERGENCY => 7,
+    );
+
+    private $minLevelIndex;
+    private $formatter;
+    private $handle;
+
+    public function __construct($minLevel = LogLevel::WARNING, $output = 'php://stderr', callable $formatter = null)
+    {
+        if (!isset(self::$levels[$minLevel])) {
+            throw new InvalidArgumentException(sprintf('The log level "%s" does not exist.', $minLevel));
+        }
+
+        $this->minLevelIndex = self::$levels[$minLevel];
+        $this->formatter = $formatter ?: array($this, 'format');
+        if (false === $this->handle = @fopen($output, 'a')) {
```

Comment:
why leave open a handle in the constructor ? Why not use `file_put_contents` which opens / closes the handle each time it is needed ?

Reply:
For performance.

Reply:
But for something that is used as a daemon, isn't this more costly to leave an handle open for a long while, rather than opening / closing it ?

A middleground could be to open the handle only on the first log call (if no logs, no need to open it) ?

Reply:
It was my first implementation, but after a discussion with @nicolas-grekas, we agreed that it costs almost nothing to open a handle for stderr.

Reply:
Ok, fair enough

Reply:
opening in the constructor gives the possibility to throw an exception in the constructor, rather that "who knows where - an nowhere fits, this is a logger".
opening lazilly is not worth it IMHO, that's a single fopen.
closing for long running process would not make them faster, just slower in fact (but that'd be negligible yes.)